streamingest: assign one partition per proc

dt opened the pull request:
Combining multiple partitions into a single proc's spec reduces the parallelism at which those partitions are subscribed and the subscriptions are buffered and ingested. Rather than make a single proc able to consume many partitions in parallel -- including buffering them separately to avoid overlapping key ranges during ingestions -- we can instead just change planning to plan one proc per partition, as each proc already has a subscription and buffer.

Release note: none.
Epic: none.

dt committed:
streamingest: assign one partition per proc

Combining multiple partitions into a single proc's spec reduces the parallelism
at which those partitions are subscribed and the subscriptions are buffered
and ingested. Rather than make a single proc able to consume many partitions
in parallel -- including buffering them separately to avoid overlapping key
ranges during ingestions -- we can instead just change planning to plan one
proc per partition, as each proc already has a subscription and buffer.

Release note: none.
Epic: none.

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/123162)
<!-- Reviewable:end -->

dt commented:
We can remove mergedSubscription after this, but I didn't do it here, since we might want to backport this since it is a pre-req for more processors and less scan parallelism per proc.

dt committed:
streamingccl/streamingest: consume all kvs in event

This switching from passing single KV events from the subscription to the
ingest processor to passing batches of KVs. This avoids needing to go back
to the channel and select for each KV, but rather for each batch of KVs.

Release note: none.
Epic: none.

dt commented:
bors r+
